### PR TITLE
[LMS-322][BUTTON] Radio Button

### PR DESCRIPTION
**Pull request by epifaniovillaro**
## Issue Link
[LMS-322](https://framgiaph.backlog.com/view/LMS-322)

## Steps to reproduce
test:http://localhost:3000/demo/components
component location: shared/components/RadioButton/index.tsx

## Affected Components / Functionalities / Page
demo/components/index.tsx

## Test Cases
_will update soon..._

## Notes
n/a

## Screenshots
image
image

### CODE REVIEW

File: client/src/shared/components/Radiobutton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import React, { Fragment } from 'react';
+
+export interface RadioButtonProps {
+  label?: string;
+  options: string[];
+  alignment?: string;
+  classname?: string;
+  onClickEvent: () => unknown;
```
Comment (escbooster12-dev) [inline client/src/shared/components/Radiobutton/index.tsx:8]: - return the selected option.
```suggestion
  onClickEvent: (query: string) => void;
```

File: client/src/shared/components/Radiobutton/index.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+const RadioButton: React.FC<RadioButtonProps> = ({ label, options, alignment, classname, onClickEvent }: RadioButtonProps) => {
+  const [chosenOption, setChosenOption] = React.useState('');
```
Comment (escbooster12-dev) [inline client/src/shared/components/Radiobutton/index.tsx:12]: - instead of `chosenOption` use `selectedOption`

Comment (escbooster12-dev) [inline client/src/shared/components/Radiobutton/index.tsx:12]: - import the use `useState` at the top

File: client/src/shared/components/Radiobutton/index.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+export interface RadioButtonProps {
+  label?: string;
+  options: string[];
+  alignment?: string;
+  classname?: string;
+  onClickEvent: () => unknown;
+}
+
+const RadioButton: React.FC<RadioButtonProps> = ({ label, options, alignment, classname, onClickEvent }: RadioButtonProps) => {
```
Comment (escbooster12-dev) [inline client/src/shared/components/Radiobutton/index.tsx:11]: import `FC` at the top

**Comment by epifaniovillaro** [comment]
[LMS-322][BUTTON] Radio Button

**Comment by epifaniovillaro** [comment]
image

File: client/src/shared/components/Radiobutton/index.tsx [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
   alignment?: string;
   classname?: string;
-  onClickEvent: () => unknown;
+  onClickEvent: (query: string) => void;
 }
 
-const Radiobutton: React.FC<Props> = ({ label, options, alignment, classname, onClickEvent }: Props) => {
-  const [chosenOption, setChosenOption] = React.useState('');
+const RadioButton: FC<Props> = ({ label, options, alignment, classname, onClickEvent }: Props) => {
+  const [selectedOption, setSelecteOption] = useState('');
```
Comment (escbooster12-dev) [inline client/src/shared/components/Radiobutton/index.tsx:13]:
```suggestion
  const [selectedOption, setSelectedOption] = useState('');
```

File: client/src/shared/components/Radiobutton/index.tsx [hunk 5]
```diff
@@ -1,18 +1,19 @@
-import React, { Fragment } from 'react';
+import React, { useState, Fragment } from 'react';
+import type { FC } from 'react';
```
Comment (escbooster12-dev) [inline client/src/shared/components/Radiobutton/index.tsx:3]: - `RadioButton`
image

**Review by Marielleregner7 — APPROVED** [review]
**APPROVED**
✅ Able to create radio button based on the Figma design.
✅ No ESLint warnings or errors.
✅ No error found, when running npm run build.

Comments: 13 (including previous [PR](https://github.com/framgia/sph-lms/pull/27) )
Test time: 35 mins
Iterations: 2

**Review by escbooster12-dev — APPROVED** [review]

**Pull request merged**